d2/src/main/java/com/linkedin/d2/discovery/stores/zk/ZooKeeperEphemeralStore.java

```diff
@@ -706,13 +747,24 @@ public void processResult(int rc, String path, Object ctx, List<String> children
       {
         case OK:
         {
+          if (_isInitialFetch) {
+            emitSDStatusInitialRequestEvent(property, true);
+            // reset initial fetch states
+            _isInitialFetch = false;
+            _initialFetchStartAt = Long.MAX_VALUE;
```

Comment:
Why are we changing this back to Long.MAX_VALUE?

Reply:
If the uri parent node is gone while we're watching it, _isInitialFetch will become true again, so that when the node is back, the first fetch will still be initial fetch. In case anywhere in the code missed to set _initialFetchStartAt to current time, wanted to make sure the last initial fetch start timestamp won't be used by mistake (the event won't be emitted since the startAt time will be greater than the current time).